Packages/ClientRuntime/Sources/Retries/SDKRetryer.swift

```diff
@@ -95,3 +100,16 @@ public class SDKRetryer: Retryer {
         }
     }
 }
+
+extension AwsCommonRuntimeKit.RetryStrategy {
+    convenience init(options: RetryOptions, eventLoopGroup: EventLoopGroup) throws {
+       try self.init(
+            eventLoopGroup: eventLoopGroup,
+            initialBucketCapacity: options.initialBucketCapacity,
+            maxRetries: options.maxRetries,
+            backOffScaleFactor: options.backOffScaleFactor,
+            jitterMode: options.jitterMode.toCRTType(),
+            generateRandom: nil // we should pass in the options.generateRandom but currently it fails since the underlying closure is a c closure
```

Comment:
This is the same behavior currently today, where we end up just passing in `nil` for this closure since we cannot convert swift closures to c closures due to constraints around c closures. 

I've talked to @waahm7 about this and filed https://github.com/awslabs/aws-crt-swift/issues/127 for CRT to expose this as a swift closure. Once that is done, we can update this here to pass in the provided swift closure.